### PR TITLE
DashboardPage: refactored styles from sass to emotion

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR refactors styles from sass to emotion.
**Which issue(s) this PR fixes**:

Relates to #32784

### CODE REVIEW

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 1]
```diff
@@ -269,10 +282,10 @@ export class DashboardPage extends PureComponent<Props, State> {
     locationService.push('/');
   };
 
-  renderSlowInitState() {
+  renderSlowInitState(dashboardLoadingstyle: string, dashboardLoadingTextStyle: string) {
```
Comment (hugohaggmark) [inline public/app/features/dashboard/containers/DashboardPage.tsx:285]: Maybe we should pass the styles object into the render functions instead? Or even better take this opportunity to refactor out smaller components from these render functions. WDYT?

File: public/app/features/dashboard/containers/DashboardPage.test.tsx [hunk 2]
```diff
@@ -67,6 +68,7 @@ function dashboardPageScenario(description: string, scenarioFn: (ctx: ScenarioCo
           cancelVariables: jest.fn(),
           templateVarsChangedInUrl: jest.fn(),
           dashboard: null,
+          theme: getTheme(),
```
Comment (hugohaggmark) [inline public/app/features/dashboard/containers/DashboardPage.test.tsx:71]: We are moving away from `emotion` to `RTL` so it would be great to convert these tests too.

**Review by hugohaggmark — COMMENTED** [review]
Looks good and works as expected 🎉  Left some comments for consideration.

File: public/app/features/dashboard/containers/DashboardPage.tsx (hunk 1 above)
Comment (thisisobate) [reply public/app/features/dashboard/containers/DashboardPage.tsx:285]: I think passing the styles object will do in this case. I don't think these render functions are worth refactoring as a standalone component because it's not really reused across other components.

File: public/app/features/dashboard/containers/DashboardPage.test.tsx (hunk 2 above)
Comment (thisisobate) [reply public/app/features/dashboard/containers/DashboardPage.test.tsx:71]: You mean the entire test file?

File: public/app/features/dashboard/containers/DashboardPage.tsx (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/containers/DashboardPage.tsx:285]: Do you prefer bigger complex classes over smaller functional components? I don't and it's not reuse I was thinking about in this case.

File: public/app/features/dashboard/containers/DashboardPage.test.tsx (hunk 2 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/containers/DashboardPage.test.tsx:71]: Yes

File: public/app/features/dashboard/containers/DashboardPage.tsx (hunk 1 above)
Comment (thisisobate) [reply public/app/features/dashboard/containers/DashboardPage.tsx:285]: Oh...I get your point now. In that case, then it makes alot of sense to refactor the render function.

File: public/app/features/dashboard/containers/DashboardPage.test.tsx (hunk 2 above)
Comment (thisisobate) [reply public/app/features/dashboard/containers/DashboardPage.test.tsx:71]: Ok

Comment (hugohaggmark) [reply public/app/features/dashboard/containers/DashboardPage.test.tsx:71]: So I started to look at this together with @thisisobate and it's quite an undertaking to migrating this to RTL so we decided to postpone this work.

Comment (thisisobate) [reply public/app/features/dashboard/containers/DashboardPage.test.tsx:71]: yes. Agreed.

**Review by hugohaggmark — APPROVED** [review]
Skilled work on this 🎉

File: public/app/features/dashboard/components/DashboardLoading/DashboardFailed.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { Alert, useStyles } from '@grafana/ui';
+import { GrafanaTheme } from '@grafana/data';
+import { DashboardInitError, AppNotificationSeverity } from 'app/types';
+import { getMessageFromError } from 'app/core/utils/errors';
+
+export interface Props {
+  initError?: DashboardInitError;
+}
+
+export const DashboardFailed: FC<Props> = ({ initError }) => {
```
Comment (torkelo) [inline public/app/features/dashboard/components/DashboardLoading/DashboardFailed.tsx:12]:
```suggestion
export const DashboardFailed = ({ initError }: Props) => {
```
We decided on frontend weekly to stop using the FC type,

File: public/sass/components/_panel_editor.scss [hunk 4]
```diff
@@ -62,3 +66,87 @@
...
+
+  .panel-container {
+    border: none;
+  }
+
+  .panel-menu-toggle,
+  .panel-menu {
+    display: none;
+  }
+}
```
Comment (torkelo) [inline public/sass/components/_panel_editor.scss:152]: why was this moved into panel editor file?

File: public/app/features/dashboard/components/DashboardLoading/DashboardFailed.tsx (hunk 3 above)
Comment (thisisobate) [reply public/app/features/dashboard/components/DashboardLoading/DashboardFailed.tsx:12]: yeah...true.
would make changes to this

File: public/sass/components/_panel_editor.scss (hunk 4 above)
Comment (thisisobate) [reply public/sass/components/_panel_editor.scss:152]: Noticed the class name isn't used in the DashboardPage so I decided to move it into the corresponding style file. Figured it would be easier for me to refactor the panel styles if they're kept in a single place.

**Review by Clarity-89 — COMMENTED** [review]
I think to fix the tests, you need to use `UnthemedDashboardPage` now, since `withTheme` return doesn't match `ShallowWrapper` Component type definition. E.g.:
```
import { UnthemedDashboardPage, mapStateToProps, Props, State } from './DashboardPage';

//...
interface ScenarioContext {
  //...
  wrapper?: ShallowWrapper<Props, State, UnthemedDashboardPage>;
 }

// On line 77
  ctx.wrapper = shallow(<UnthemedDashboardPage {...props} />);
```

Also remember to update the snapshots.

File: public/sass/components/_panel_editor.scss (hunk 4 above)
Comment (torkelo) [inline public/sass/components/_panel_editor.scss:152]: can you revert these changes and move them pack to __dashboard.scss ?

Comment (torkelo) [reply public/sass/components/_panel_editor.scss:152]: or move them to a __panel.scss

Comment (thisisobate) [reply public/sass/components/_panel_editor.scss:152]: Alright 👍

**Comment by thisisobate** [comment]
> I think to fix the tests, you need to use `UnthemedDashboardPage` now, since `withTheme` return doesn't match `ShallowWrapper` Component type definition. E.g.:
> 
> ```
> import { UnthemedDashboardPage, mapStateToProps, Props, State } from './DashboardPage';
> 
> //...
> interface ScenarioContext {
>   //...
>   wrapper?: ShallowWrapper<Props, State, UnthemedDashboardPage>;
>  }
> 
> // On line 77
>   ctx.wrapper = shallow(<UnthemedDashboardPage {...props} />);
> ```
> 
> Also remember to update the snapshots.

Alright. Thanks!

File: public/sass/components/_panel_editor.scss (hunk 4 above)
Comment (thisisobate) [reply public/sass/components/_panel_editor.scss:152]: ok

Comment (thisisobate) [reply public/sass/components/_panel_editor.scss]: I couldn't find `_panel.scss` so I moved it back to `_dashboard.scss`
